src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/SqlDelegatedTransaction.cs

```diff
@@ -421,7 +454,7 @@ internal void TransactionEnded(Transaction transaction)
         private SqlInternalConnection GetValidConnection()
         {
             SqlInternalConnection connection = _connection;
-            if (null == connection)
+            if (null == connection && Transaction.TransactionInformation.Status != TransactionStatus.Aborted)
```

Comment:
This new logic seems odd to me. So if the connection is null but we have an active transaction, we should now return a null connection instead of throwing?

Reply:
If the connection is null but we have an "Aborted" transaction, then we return a null connection.
If the connection is null but we have an "Active" transaction, then we throw exception.